packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -277,44 +418,111 @@ class _DatePickerDialog extends StatefulWidget {
   /// The initial display of the calendar picker.
   final DatePickerMode initialCalendarMode;
 
+  /// The error text displayed if the entered date is not in the correct format.
   final String? errorFormatText;
 
+  /// The error text displayed if the date is not valid.
+  ///
+  /// A date is not valid if it is earlier than [firstDate], later than
+  /// [lastDate], or doesn't pass the [selectableDayPredicate].
   final String? errorInvalidText;
 
+  /// The hint text displayed in the [TextField].
+  ///
+  /// If this is null, it will default to the date format string. For example,
+  /// 'mm/dd/yyyy' for en_US.
   final String? fieldHintText;
 
+  /// The label text displayed in the [TextField].
+  ///
+  /// If this is null, it will default to the words representing the date format
+  /// string. For example, 'Month, Day, Year' for en_US.
   final String? fieldLabelText;
 
   @override
   _DatePickerDialogState createState() => _DatePickerDialogState();
 }
 
-class _DatePickerDialogState extends State<_DatePickerDialog> {
+// A restorable data structure to save and restore dates. It only accounts
```

Comment:
@darrenaustin For the date picker, only day, month, and year are important, right?

@goderbauer I decided to make a `_RestorableDate` property because it felt cleaner than using three (or perhaps more) `RestorableInt`s everywhere in the widget and then having to convert between a `DateTime` and those three ints. What do you think?

Reply:
+1 to having a RestorableDateTime (see my other comments)